worker.go

```diff
@@ -70,6 +81,14 @@ func (w *Worker) Fetch(url string) (records, error) {
 		if err == nil {
 			break
 		}
+        
+        if w.query.ErrorValue != "" {
+            rec := make(map[string]interface{})
+            rec["error"] = w.query.ErrorValue
+            var recs []record
+            recs = append(recs, rec)
+            w.SetMetrics(recs)
```

Comment:
This can be a one-liner and removes the additional internal buffer of the slice due to an append call.

```go
w.SetMetrics([]record{
    map[string]interface{}{
        "error": w.query.ErrorValue,
    },
})
```